### PR TITLE
Fix Markdown rendering within summary/detail

### LINKED ISSUE #1878: EIP1474 Does not render correctly on main EIPs website
See: https://eips.ethereum.org/EIPS/eip-1474

It appears the markup on that page is interpreting all of the *collapsible* sections as being nested.

I would advocate for removing the collapse functionality in favor of a table of contents and maybe links in each section to take you back to the TOC.

### PR DESCRIPTION
**Pull request by axic**
Tries to fix the rendering of https://eips.ethereum.org/EIPS/eip-1474

Idea from: https://github.com/gettalong/kramdown/issues/155#issuecomment-339402782

Fixes #1878.

### CODE REVIEW

**Comment by axic** [comment]
This actually triggered two ERCs which have bad formatting:
```
- ./_site/EIPS/eip-1154.html
  *  171:14: ERROR: Unexpected end tag : dd (line 171)
  *  172:12: ERROR: Unexpected end tag : dd (line 172)
  *  173:10: ERROR: Unexpected end tag : dd (line 173)
  *  174:8: ERROR: Unexpected end tag : dd (line 174)
- ./_site/EIPS/eip-1620.html
  *  linking to internal hash #log-confirm-update that does not exist (line 228)
     <a href="#log-confirm-update">LogConfirmUpdate</a>
  *  linking to internal hash #log-create that does not exist (line 195)
     <a href="#log-create">LogCreate</a>
  *  linking to internal hash #log-execute-update that does not exist (line 230)
     <a href="#log-execute-update">LogExecuteUpdate</a>
  *  linking to internal hash #log-redeem that does not exist (line 217)
     <a href="#log-redeem">LogRedeem</a>
  *  linking to internal hash #log-revoke-update that does not exist (line 241)
     <a href="#log-revoke-update">LogRevokeUpdate</a>
  *  linking to internal hash #log-withdraw that does not exist (line 206)
     <a href="#log-withdraw">LogWithdraw</a>
```

**Comment by axic** [comment]
I've fixed some of these "issues" locally, but actually realised that for some reason (probably due to `Gemfile.lock`) the github-pages packages is stuck at 179, while the current release 198. I'm sure the new version could fix a lot of rendering problems as it brings in newer kramdown.

**Comment by Arachnid** [comment]
Do you want to pursue this PR further?

**Comment by axic** [comment]
Yes. Need to figure out how I can convince my bundler to update the versions.

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - File .gitignore is not an EIP
 - File .travis.yml is not an EIP
 - File Gemfile is not an EIP
 - File Gemfile.lock is not an EIP
 - File _config.yml is not an EIP
 - EIP 1057 requires approval from one of (@ifdefelse)
 - EIP 1102 requires approval from one of (@bitpshr)
 - EIP 1123 requires approval from one of (@djrtwo, @pipermerriam, @njgheorghita, @gnidan)
 - EIP 1474 requires approval from one of (@bitpshr)
 - EIP 1484 requires approval from one of (shanehampton1@gmail.com, andychorlian@gmail.com, anurag.angara@gmail.com, @noahzinsmeister)
 - EIP 1620 requires approval from one of (hello@paulrberg.com, @paulrberg)
 - EIP 1775 requires approval from one of (@danfinlay, @bunjin)

File: _config.yml [hunk 1]
```diff
@@ -36,6 +36,12 @@ twitter:
 # Build settings
 markdown: kramdown
 theme: minima
+kramdown:
+  parse_block_html: false
+  # This is the default, but be explicit as some EIPs depend on it
+  auto_ids: true
+  # This is to ensure more determistic behaviour
+  auto_id_stripping: true
```
Comment (axic) [inline _config.yml]: Link generation is explained here: https://kramdown.gettalong.org/converter/html.html#auto-ids

Newer version of kramdown changed the rules, hence the changes to existing EIPs.

File: _config.yml [hunk 2]
```diff
@@ -36,6 +36,12 @@ twitter:
 # Build settings
 markdown: kramdown
 theme: minima
+kramdown:
+  parse_block_html: false
```
Comment (axic) [inline _config.yml]: Turned this to `true` means that markdown parsing is enabled within `<details>` blocks. However it is not working well. The closing HTML tags are not parsed.

This introduces problems with other EIPs which use the `details` block properly (820, 1820, etc.): only with a larger code snippet and not with embedded markdown.

The resolution is to change EIP-1474 instead, which was the culprit triggering this entire PR.

**Comment by axic** [comment]
@Arachnid should be ready to merge.

File: EIPS/eip-1123.md [hunk 3]
```diff
@@ -1397,8 +1393,6 @@ The `name` field defines which compiler was used in compilation.
 </table>
 
 
-<div id="version-version-1"></div>
```
Comment (axic) [inline EIPS/eip-1123.md]: Kramdown generates proper links out the headers now, no need for these. If left here htmlproofer complains about duplicate IDs. Manually verified they point to the right header.

File: _config.yml (hunk 2 above)
Comment (bitpshr) [reply _config.yml]: I can update EIP-1474 accordingly.

Comment (axic) [reply _config.yml]: I already did in this PR.

**Comment by axic** [comment]
@Arachnid any comments on merging this?

**Comment by axic** [comment]
@Arachnid @nicksavers any objections merging this?

**Pull request merged**